tensorflow/lite/micro/tools/ci_build/test_size.sh

```diff
@@ -14,38 +14,63 @@
 # limitations under the License.
 # ==============================================================================
 #
-# Build a TFLite micro test binary and capture its size.
-
+# This script builds a TFLite micro test binary and compare the size difference
+# between this binary and that same binary from the main repo.
+# If the optional argument string "error_on_memory_increase" is provided as the
+# script input, the script will error exit on any memory increase.
+# If no argument is provided, the script produce a size comparison report.
 set -e
 
-SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
-ROOT_DIR=${SCRIPT_DIR}/../../../../..
-cd "${ROOT_DIR}"
+source tensorflow/lite/micro/tools/ci_build/helper_functions.sh
+
+# Utility function to build a target and return its path back to caller through
+# a global variable __BINARY_TARGET_PATH.
+# The caller is expected to store this  __BINARY_TARGET_PATH back to its local
+# variable if it needs to use the generated binary target with path later on.
+__BINARY_TARGET_PATH=
+function build_target() {
+  local binary_target=$1
+  local build_type=$2
+  local target=$3
+  local target_arch=$4
+  readable_run make -f tensorflow/lite/micro/tools/make/Makefile clean
+  readable_run make -f tensorflow/lite/micro/tools/make/Makefile third_party_downloads
+  readable_run make -j8 -f tensorflow/lite/micro/tools/make/Makefile build build_type=${build_type} TARGET=${target} TARGET_ARCH=${target_arch} ${binary_target}
 
+  # Return the relative binary with path and name.
+  __BINARY_TARGET_PATH="tensorflow/lite/micro/tools/make/gen/${target}_${target_arch}_${build_type}/bin/${binary_target}"
```

Comment:
one recommendation for returning values from bash functions is to echo them.

https://www.linuxjournal.com/content/return-values-bash-functions